File: kotlin/src/main/kotlin/com/di/kotlin/no/di/UserService.kt
package com.di.kotlin.no.di

import com.di.kotlin.IUserService
import com.di.kotlin.model.User

class UserService : IUserService {
    private val userRepository = UserRepository()
    private val validationsService = ValidationsService()
    private val logger = Logger()

    override fun create(nombre: String, apellido: String): User {
        val valid = this.validationsService.validateIdentidad(nombre, apellido)
        if (!valid) {
            throw Exception("El nombre o el apellido no son validos")
        }
        val user = this.userRepository.createUser(nombre, apellido)
        this.logger.log("User created: $user")
        return user
    }

    override fun findById(id: Long) : User? {
        return this.userRepository.findById(id)
    }
}

File: nodejs/src/conDependencyInjection/conLibreria/UserService.js
export class UserService {
    constructor(opts) {
        this.userRepository = opts.userRepository;
        this.logger = opts.logger;
        this.validationsService = opts.validationsService;
    }

    create(nombre, apellido) {
        const valid = this.validationsService.validateIdentidad(nombre, apellido);
        if (!valid) {
            throw Error("El nombre o el apellido no son validos");
        }
        const user = this.userRepository.create(nombre, apellido);
        this.logger.log(`User created: ${JSON.stringify(user,null,2)}`);
        return user
    }

    findById(id) {
        return this.userRepository.findById(id);
    }
}